src/test/flutter_create_tests/flutter_create.test.ts

```diff
@@ -14,106 +14,63 @@ describe("flutter", () => {
 
 	it("created a basic default project", async () => {
 		const basicProjectFolder = fsPath(vs.workspace.workspaceFolders![0].uri);
-		const expectedString = "title: 'Flutter Demo'";
-		const mainFile = path.join(basicProjectFolder, "lib", "main.dart");
-		// Creating the sample may be a little slow, so allow up to 60 seconds for it.
-		logger.info("Waiting for file to exist", LogCategory.CI);
-		await waitForResult(() => fs.existsSync(mainFile), "lib/main.dart did not exist", 100000);
-		// Wait for up to 10 seconds for the content to match, as the file may be updated after creation.
-		logger.info("Waiting for content match", LogCategory.CI);
-		await waitForResult(() => {
-			const contents = fs.readFileSync(mainFile);
-			return contents.indexOf(expectedString) !== -1;
-		}, undefined, 10000, false); // Don't throw on failure, as we have a better assert below that can include the contents.
-
-		const contents = fs.readFileSync(mainFile);
-		if (contents.indexOf(expectedString) === -1)
-			assert.fail(`Did not find "${expectedString}'" in the sample file:\n\n${contents}`);
+		const pubspecFile = path.join(basicProjectFolder, "lib", "main.dart");
+
+		await projectFileContainsExpectedString(pubspecFile, "title: 'Flutter Demo'");
 	});
 
 	it("created a sample project", async () => {
 		const sampleProjectFolder = fsPath(vs.workspace.workspaceFolders![1].uri);
-		// const expectedString = "Flutter code sample for material.IconButton.1";
-		const expectedString = "This sample shows an `IconButton` that";
-		const mainFile = path.join(sampleProjectFolder, "lib", "main.dart");
-		// Creating the sample may be a little slow, so allow up to 60 seconds for it.
-		logger.info("Waiting for file to exist", LogCategory.CI);
-		await waitForResult(() => fs.existsSync(mainFile), "lib/main.dart did not exist", 100000);
-		// Wait for up to 10 seconds for the content to match, as the file may be updated after creation.
-		logger.info("Waiting for content match", LogCategory.CI);
-		await waitForResult(() => {
-			const contents = fs.readFileSync(mainFile);
-			return contents.indexOf(expectedString) !== -1;
-		}, undefined, 10000, false); // Don't throw on failure, as we have a better assert below that can include the contents.
-
-		const contents = fs.readFileSync(mainFile);
-		if (contents.indexOf(expectedString) === -1)
-			assert.fail(`Did not find "${expectedString}'" in the sample file (${mainFile}):\n\n${contents}`);
+		const pubspecFile = path.join(sampleProjectFolder, "lib", "main.dart");
```

Comment:
nit: variable name - this isn't a pubspec

Reply:
(I'll rename this, as landing this now to prepare a pre-release build)

Reply:
D'oh - thanks!